source/content/guides/composer-convert.md

```diff
@@ -1,156 +1,301 @@
+The steps here ensure that any modules and themes from [drupal.org](drupal.org) are in the `composer.json` `require` list.
 
-```bash{outputLines:2-4}
-find . -maxdepth 2 -name '*.info.yml' \
-  -exec basename '{}' '.info.yml' ';' \
-  -exec grep -e '^version' '{}' ';' \
-  -exec echo ';'
-```
+Once Composer is aware of all the contributed code, you'll be able to run `composer upgrade` from within the directory to have Composer upgrade all the contributed code automatically.
 
-This will list each module followed by the version of that module that is installed.
+Begin by reviewing the existing site's code. Check for contributed modules in `/modules`, `/modules/contrib`, `/sites/all/modules`, and `/sites/all/modules/contrib`.
 
-You can add these modules to your new codebase using Composer by running the following for each module in the `$site-composer` directory:
+1. When reviewing the site, take stock of exactly what versions of modules and themes you depend on. One way to do this is to change to run the `pm:projectinfo` Drush command from within a contributed modules folder (e.g. `/modules`, `/themes`, `/themes/contrib`, `/sites/all/themes`, `/sites/all/themes/contrib`, etc.).
 
-```{promptUser:user}
-composer require drupal/MODULE_NAME:^VERSION
-```
+  This command works on Drush 8. If you're using Drush 9, use `pm:list` or refer to [Drush Commands](https://drushcommands.com/drush-9x/pm/pm:projectinfo/):
 
-Where `MODULE_NAME` is the machine name of the module in question, and `VERSION` is the version of that module your site is currently using. Composer may pull in a newer version than what you specify, depending upon what versions are available. You can read more about the caret (`^`) [in the documentation for Composer](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) itself.
+  ```bash{promptUser:user}
+  terminus drush $SITE.dev -- pm:projectinfo --fields=name,version --format=table
+  ```
 
-If you get the following error:
+  This will list each module followed by the version of that module that is installed.
 
-```none
-[InvalidArgumentException]
-Could not find a version of package drupal/MODULE_NAME matching your minimum-stability (stable). Require it with an explicit version constraint allowing its desired stability.
-```
+1. You can add these modules to your new codebase using Composer by running the following for each module in the `$SITE-composer` directory:
+
+  ```bash{promptUser:user}
+  composer require drupal/MODULE_NAME:^VERSION
+  ```
+
+  Where `MODULE_NAME` is the machine name of the module in question, and `VERSION` is the version of that module the site is currently using. Composer may pull in a newer version than what you specify, depending upon what versions are available. You can read more about the caret (`^`) in the [Composer documentation](https://getcomposer.org/doc/articles/versions.md#caret-version-range-).
```

Comment:
At one point we had an explanation about versions strings.

Old-style Drupal version strings:
```
Chaos Tools (ctools)  8.x-3.4
```
Drop the `8.x-` to convert this into `^3.4`

Semver version strings:
```
Devel (devel)  4.1.1
```
Use this version directly, e.g. `^4.1.1`

Reply:
I've added this a subpoints in https://github.com/pantheon-systems/documentation/pull/6145/files#diff-de5e95860ba4c809f9e43c75a3ff0cd70ad67998ae04beec2df43d8d52958415R200-R216